resources/js/Organisms/Form.tsx

```diff
@@ -48,15 +49,49 @@ const Form: React.FC<IFormProps> = ({ action, initialValues, render }) => {
   };
 
   const updateState = (target: HTMLInputElement) => {
-    const { name, value } = target;
-    setValues({ ...values, [name]: value });
+    const { name, value,type } = target;
+    if(type==='checkbox'){
+      let newSelection: any[];
+      if(values[name].indexOf(value) > -1) 
+      {
+        newSelection = values[name].filter(s => s !== value)
+      }else 
+      {
+        newSelection = [...values[name], value];
+      }
+      setValues({ ...values, [name]: newSelection }); 
+    }else
+      {
+        setValues({ ...values, [name]: value });
+      }
+
+    if(haveErrors(errors))
+      {
+        validateForm();
+      }
   };
 
   const validateForm = (): boolean => {
-    // TODO - validate form
-    // (set errors? for now just set as empty object)
-    setErrors({});
-    return true;
+    let newErrors:IErrors = errors;
+    let formisValid:boolean = true;
+    newErrors.fullname ='';
+    newErrors.email ='';
+
+    if (!values["fullname"] || values["fullname"].trim().length==0) 
+      {
+        newErrors.fullname ='Please enter your fullname';
+        formisValid=false;
+      }
+
+    let validEmail = /^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/; 
+    if (!validEmail.test(values["email"])) 
+      {
+        errors.email = 'Please enter valid email id';
+        formisValid = false;
+      }  
+      
+    setErrors({...errors,["fullname"]:newErrors.fullname,["email"]:newErrors.email});  
```

Comment:
I think this would be a bit easier to read with a few spaces and without the `["..."]` around the keys - like this:
```suggestion
    setErrors({ ...errors, fullname: newErrors.fullname, email: newErrors.email });  
```

This is equivalent to what you wrote, so if you find it more readable with th `["` then feel free to leave it as-is 😄